src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions_Negate.cs

```diff
@@ -272,6 +273,33 @@ private static SyntaxNode GetNegationOfIsPatternExpression(SyntaxNode isExpressi
             return generator.LogicalNotExpression(isExpression);
         }
 
+        private static SyntaxNode GetNegationOfRelationalPattern(
+            SyntaxNode expressionNode,
+            SyntaxGeneratorInternal generatorInternal,
+            SpecialType? patternValueType)
+        {
+            if (patternValueType is SpecialType specialType && specialType.IsNumericType())
+            {
+                // If we know the value is numeric, we can negate the relational operator.
+                // This is not valid for non-numeric value since they never match a relational pattern.
+                // Similarly, it's not valid for nullable values, since null never matches a relational pattern.
+                // As an example, `!(new object() is < 1)` is equivalent to `new object() is not < 1` but not `new object() is >= 1`.
+                var syntaxFacts = generatorInternal.SyntaxFacts;
+                syntaxFacts.GetPartsOfRelationalPattern(expressionNode, out var operatorToken, out var expression);
+                syntaxFacts.TryGetPredefinedOperator(operatorToken, out var predefinedOperator);
+                return predefinedOperator switch
+                {
+                    PredefinedOperator.LessThan => generatorInternal.GreaterThanEqualsPattern(expression),
+                    PredefinedOperator.LessThanOrEqual => generatorInternal.GreaterThanPattern(expression),
+                    PredefinedOperator.GreaterThan => generatorInternal.LessThanEqualsPattern(expression),
+                    PredefinedOperator.GreaterThanOrEqual => generatorInternal.LessThanPattern(expression),
+                    _ => throw ExceptionUtilities.UnexpectedValue(predefinedOperator)
```

Comment:
this is likely not safe.  i think we allow more relational operators in error recovery scenarios (this would alos be a good test).  e.g. `switch x {  == 0 => ... }`.   In this case, we should just add the 'not', not crash.

Reply:
I noticed `SyntaxFactory.RelationalPattern` supports `==` and `!=` as well, but couldn't figure out why. I can't seem to find a situation where they're valid relational patterns. For example, the `switch` expression you posted above doesn't compile ([CS1525](https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs1525)). I don't know what error recovery scenarios are, nor how to test them.

> In this case, we should just add the 'not', not crash.
That makes sense to me, however. 😄

Reply:
>  I don't know what error recovery scenarios are, nor how to test them.

Error recovery scenarios are exactly that.  Code with errors in them.  Roslyn should not ever crash in these cases, regardless of how illegal the code is.  The way to write the test is to just use those other relational operators, not just the specific ones taht are legal.  Thanks!

Reply:
Thanks for explaining. I added a test for `is ==`. (For now it defaults to negating to `is not ==`. We could have it negate to `is !=`, but I don't think it makes much of a difference.)

Reply:
As it's an error case, i'm totally fine with the simple behavior.  Thanks!